wsd/Storage.hpp

```diff
@@ -286,19 +237,19 @@ class StorageBase
         std::string _reason;
     };
 
-    /// The state of an asynchronous upload request.
-    class AsyncUpload final
+    /// The state of an asynchronous request.
+    template <typename TResult> class AsyncRequest final
     {
     public:
-        enum class State
-        {
+        STATE_ENUM(
+            State,
             None, //< No async upload in progress or isn't supported.
             Running, //< An async upload request is in progress.
             Error, //< Failed to make an async upload request or timed out, no UploadResult.
             Complete //< The last async upload request completed (regardless of the server's response).
-        };
```

Comment:
I also prefer CamelCase for these constants (above all capital though)

Reply:
We have a few all-caps constants. That's probably the influence of http status codes, and the fact that these codes were defined in Poco and we ported them over.